pkg/k8s/slim/k8s/api/core/v1/types.go

```diff
@@ -566,6 +566,14 @@ type ServiceSpec struct {
 	// +optional
 	SessionAffinity ServiceAffinity `json:"sessionAffinity,omitempty" protobuf:"bytes,7,opt,name=sessionAffinity,casttype=ServiceAffinity"`
 
+	// Only applies to Service Type: LoadBalancer
+	// LoadBalancer will get created with the IP specified in this field.
+	// This feature depends on whether the underlying cloud-provider supports specifying
+	// the loadBalancerIP when a load balancer is created.
+	// This field will be ignored if the cloud-provider does not support the feature.
+	// +optional
+	LoadBalancerIP string `json:"loadBalancerIP,omitempty" protobuf:"bytes,8,opt,name=loadBalancerIP"`
```

Comment:
This field is used when users themselves want to request a specific LB IP. It is _not entirely_ required, but it is checked by the MetalLB logic.

Commit msg:

>This field is crucial for BGP integration with MetalLB because a user
can perfer an LB IP by setting this field, rather than allocate a random
IP for an IP pool.